src/particles/elements/mixin/lineartransport.H

```diff
@@ -24,27 +24,35 @@ namespace impactx::elements::mixin
 {
     /** This is a helper class for lattice elements that can be expressed as linear transport maps.
      */
+    template<typename T_Element>
     struct LinearTransport
     {
-        /** ...
-         */
-        LinearTransport (
-        )
-        {
-        }
-
-        //LinearTransport () = default;
+        LinearTransport () = default;
         LinearTransport (LinearTransport const &) = default;
         LinearTransport& operator= (LinearTransport const &) = default;
         LinearTransport (LinearTransport&&) = default;
         LinearTransport& operator= (LinearTransport&& rhs) = default;
 
         ~LinearTransport () = default;
 
-        // 6x6 linear transport map
-        using Map6x6 = amrex::SmallMatrix<amrex::ParticleReal, 6, 6, amrex::Order::F, 1>;
-        // note: for most elements, R is returned by a member function. Some store it also internally as a member.
-        // Map6x6 m_transport_map; ///< linearized map
+        /** Linear push of the covariance matrix through an element
+         *
+         * @param cm covariance matrix
+         */
+        AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+        void
+        operator() (Map6x6 & cm) const
+        {
+            static_assert(
+                std::is_base_of_v<LinearTransport, T_Element>,
+                "LinearTransport can only be used as a mixin class!"
+            );
+
+            // small trick to force every derived class has to implement a method transport_map
+            // (w/o using a purely virtual function)
+            T_Element& element = *static_cast<T_Element*>(this);
+            cm *= element.transport_map(cm);
```

Comment:
Is this logic right?  The function `transport_map` in each element takes the reference particle data as input, but this appears to call `transport_map` with the covariance matrix `cm` as input.  Also, if the intent is to push the covariance matrix cm by the matrix `R`, then the math should look like _cm = R * cm * R^T_, which appears also `PushAll.H`.

Reply:
Oh, that is outdated. I only fixed it in `ImpactX.cpp`.